complete-step-by-step-guide.md

```diff
@@ -0,0 +1,61 @@
+* Vault pki: manages the PKI elements for Vault and it's clients
+
+## Setting up Vault for Kubernetes auth and AWS credentials
+We want a Vault server able to provide AWS roles credentials to specific kube SA. Vault configuration will live in its own repo, and will be applied to Vault using terraform-applier.
+
+An example setup can be found [here](https://github.com/utilitywarehouse/kubernetes-manifests/tree/master/exp-1-aws/sys-vault).
+
+### Deploying vault
+* Setup necessary [cluster wide permissions](https://github.com/utilitywarehouse/vault-manifests/tree/master/example/cluster-wide) for Vault
+* Set up [Vault](https://github.com/utilitywarehouse/vault-manifests/tree/master/example/vault-namespace) in a new dedicated namespace
+### Creating resources needed to configure Vault
+We need to create some resources on kube and AWS for Vault to use, and then configure Vault to use those resources.
+#### Kubernetes authentication resources
+* To allow pods to login into Vault using their SA, Vault needs to ask kube who those SA are. Vault needs a SA with `system:auth-delegator` permission, that would allow it to do authentication checks with the SA tokens trying to login.
+* An [example](https://github.com/utilitywarehouse/kubernetes-manifests/blob/master/exp-1-aws/sys-vault/terraform-applier.yaml) of the SA and secret needed, living in Vault's namespace
+* An [example](https://github.com/utilitywarehouse/kubernetes-manifests/blob/master/exp-1-aws/kube-system/05-auth-vault.yaml) with the token-authentication cluster role binding granting auth-delegator to the SA
+
+#### AWS secrets resources
+* The way Vault will get credentials for the client apps is by assuming roles. Vault just needs an IAM user with no permissions, since permission to assume the roles are granted in the roles themselves ([example](https://github.com/utilitywarehouse/terraform/blob/master/aws/dev/sys-vault-exp-1/credentials-provider.tf))
+#### Terraform applier state backend
+* We are using S3 backend, so we need to create the bucket and a user with permissions to access it ([example](https://github.com/utilitywarehouse/terraform/blob/master/aws/dev/sys-vault-exp-1/terraform-state.tf))
+
+### Creating Vault's backends configuration
+* Create a repo to host the vault terraform configuration [example](https://github.com/utilitywarehouse/sys-vault-terraform) and folders for each kubernetes cluster
+* Configure the [kubernetes auth backend](https://github.com/utilitywarehouse/sys-vault-terraform/blob/master/exp-1-aws/backends/kubernetes-auth-method.tf)
+* Configure the [aws secret backend](https://github.com/utilitywarehouse/sys-vault-terraform/blob/master/exp-1-aws/backends/aws-secrets-engine.tf)
+
+### Deploy terraform applier
+* Setup [terraform applier](https://github.com/utilitywarehouse/terraform-applier/tree/master/manifests/example) in vault's namespace, and configure it to sync with your vault terraform configuration repo
```

Comment:
It might be worth adding a link to the `sys-vault` patch here as a more copy-paste example of what environment variables need to be patched:
```suggestion
* Setup [terraform applier](https://github.com/utilitywarehouse/terraform-applier/tree/master/manifests/example) in vault's namespace, and configure it to sync with your vault terraform configuration repo ([example](https://github.com/utilitywarehouse/kubernetes-manifests/blob/master/exp-1-aws/sys-vault/terraform-applier-patch.yaml))
```